fastchat/serve/inference.py

```diff
@@ -70,6 +70,13 @@ def generate_stream(
     input_ids = tokenizer(prompt).input_ids
     input_echo_len = len(input_ids)
     output_ids = list(input_ids)
+    
+    if len(input_ids) + max_new_tokens > context_len:
+        raise ValueError(f"This model's maximum context length is {context_len} tokens. "
+                         f"However, you requested {max_new_tokens + len(input_ids)} tokens "
+                         f"({len(input_ids)} in the messages, "
+                         f"{max_new_tokens} in the completion). "
+                         f"Please reduce the length of the messages or completion.")
```

Comment:
Can we delete this? In L86 we automatically truncate the old history. If we raise exception here, our chat demo will not work after a long conversation

Reply:
Ok, here the length detection is the behavior of OpenAI's. I think length checking is necessary. Although the old history truncation should be done in the demo, here I can first move this detection to openai_server to make the demo work.